common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/NonceRedirectHandler.kt

```diff
@@ -0,0 +1,70 @@
+package com.microsoft.identity.common.internal.ui.webview.challengehandlers
+
+import android.app.Activity
+import androidx.annotation.VisibleForTesting
+import com.microsoft.identity.common.CommonPrtCredentialHolder
+import com.microsoft.identity.common.adal.internal.util.StringExtensions
+import com.microsoft.identity.common.java.util.StringUtil
+import com.microsoft.identity.common.logging.Logger
+import java.net.MalformedURLException
+import java.net.URL
+
+/**
+ * Handler for processing nonce from redirect and attaching new prt credential header on web view.
+ */
+class NonceRedirectHandler {
+    private val TAG = NonceRedirectHandler::class.java.simpleName
+
+    private fun getPrtHeader(requestHeaders: HashMap<String, String>): String? {
+        return requestHeaders["x-ms-RefreshTokenCredential"]
+    }
+
+    // Updates the headers by attaching a new refresh token credential header (Generated using the new nonce).
+    fun getHeadersWithNewRefreshTokenCredential(
+        requestHeaders: HashMap<String, String>,
```

Comment:
HashMap are passed by reference, the function can modify the contents of the HashMap without needing to return a new object.